services/rest/websocket/ack.go

```diff
@@ -0,0 +1,38 @@
+package websocket
+
+import (
+	"time"
+
+	"github.com/odpf/raccoon/metrics"
+	"github.com/odpf/raccoon/serialization"
+	"github.com/odpf/raccoon/services/rest/websocket/connection"
+)
+
+var AckChan = make(chan AckInfo)
+
+type AckInfo struct {
+	MessageType     int
+	RequestGuid     string
+	Err             error
+	Conn            connection.Conn
+	serializer      serialization.SerializeFunc
+	TimeConsumed    time.Time
+	AckTimeConsumed time.Time
+}
+
+func AckHandler(ch <-chan AckInfo) {
+	for c := range ch {
+		ackTim := time.Since(c.AckTimeConsumed)
+		metrics.Timing("ack_event_rtt_ms", ackTim.Milliseconds(), "")
+
+		tim := time.Since(c.TimeConsumed)
+		if c.Err != nil {
+			metrics.Timing("event_rtt_ms", tim.Milliseconds(), "")
```

Comment:
@AkbaraliShaikh What is the diff between `ack_event_rtt_ms ` and `event_rtt_ms `? It will be confusing if both are different ones. Also let's have a consistent naming of metrics.

Reply:
`event_rtt_ms` -> is used to track the total time taken for the event (from consume -> response)
`ack_event_rtt_ms` -> is used to track the total time taken for only acknowledging (time spent in the ack channel).